yang/frr-bgp.yang

```diff
@@ -0,0 +1,1248 @@
+             instance.";
+          list afi-safi {
+            key "afi-safi-name";
+            description
+              "AFI, SAFI configuration available for the
+               neighbour or group.";
+            uses mp-afi-safi-config;
+
+            uses mp-all-afi-safi-list-contents;
+          }
+        }
+      }
+
+      container neighbors {
+        description
+          "Configuration for BGP neighbors.";
+        list neighbor {
+          key "remote-address";
+          description
+            "List of BGP neighbors configured on the local system,
+             uniquely identified by remote IPv[46] address.";
+          leaf remote-address {
+            type inet:ip-address;
+            description
+              "The remote IP address of this entry's BGP peer.";
+          }
+
+          leaf local-interface {
+            type frr-interface:interface-ref {
+              require-instance false;
+            }
+            description
+              "The local interface for the TCP connection between
+               the BGP peers.";
+          }
+
+          leaf local-port {
+            type inet:port-number {
+              range "0..65535";
+            }
+            description
+              "The local port for the TCP connection between
+               the BGP peers.";
+          }
+
+          leaf peer-group {
+            type leafref {
+              path "../../../peer-groups/peer-group/peer-group-name";
+            }
+            description
+              "The peer-group with which this neighbor is associated.";
+          }
+
+          uses neighbor-group-config;
+        }
+
+        list unnumbered-neighbor {
+          key "interface";
+          description
+            "List of BGP neighbors configured on the local system,
+             uniquely identified by interfaces.";
+          leaf interface {
+            type frr-interface:interface-ref {
+              require-instance false;
+            }
+            description
+              "The local interface of this entry's BGP peer.";
+          }
+
+          leaf v6only {
+            type boolean;
+            default "false";
+            description
+              "When set to 'true' it will create a neighbor with v6
+               link local only.";
+          }
+
+          leaf peer-group {
+            type leafref {
+              path "../../../peer-groups/peer-group/peer-group-name";
+            }
+            description
+              "The peer-group with which this neighbor is associated.";
+          }
+
+          uses neighbor-group-config;
+        }
+      }
+
+      container peer-groups {
+        description
+          "Configuration for BGP peer-groups.";
+        uses bgp-peer-group-list;
+      }
+    }
+  }
+
+  augment "/frr-rt:routing/frr-rt:control-plane-protocols/frr-rt:control-plane-protocol/bgp/global/afi-safis/afi-safi/ipv4-unicast" {
+    list static-route {
```

Comment:
This is a bit confusing. What does it mean to have a static route under an AFI/SAFI? Is this intended to hold redistributed routes or something? Or is this the equivalent of a network statement?

Reply:
It's for network config. I have renamed to network-config alias.